### ISSUE TITLE
Annotation Issue: Unnecessary Key Nodes

### ISSUE THREAD

**Issue by minghchen** [iMeanAI/WebCanvas#29]
The current labeled step evaluation regarding URL matching from searches only affirms the URLs found through on-site searches, rather than those accessed via Google searches.

For example, for the first task in the development set ("Browse best selling black hoodies in mens size Big and Tall that is between $25 and $50 in kohls"), since the initial URL is empty, the LLM Agent often resorts to Google search:

When using Google search, the accessed URL is: "https://www.google.com/search?q=best%20selling%20black%20hoodies%20mens%20Big%20and%20Tall%20$25%20to%20$50%20site:kohls.com". By clicking the first link, the Agent can reach: "https://www.kohls.com/catalog/mens-big-tall-hoodies-sweatshirts-tops-clothing.jsp?CN=Gender:Mens+SizeRange:Big%20%26%20Tall+Silhouette:Hoodies%20%26%20Sweatshirts+Category:Tops+Department:Clothing"

However, the labeled data indicates that one needs to access the URL through Kohl's search function: "https://www.kohls.com/search.jsp?submit-search=web-regular&search=mens+black+hoodie&kls_sbp=34524031611978259241165260194179142249"

```
{
  "match_function_name": "url_semantic_match",
  "content": {
  "key": "search",
  "reference_answer": "Decide whether are searching for mens black hoodie",
  "url": "https://www.kohls.com/search.jsp?submit-search=web-regular&search=mens+black+hoodie&kls_sbp=34524031611978259241165260194179142249"
  }
}
```

In the matching method, the key is "search", while the URL accessed via Google search does not include the "search" keyword. This ultimately leads to the agent's attempt to complete the task being considered a failure. However, both search methods can lead to the desired website for the task, indicating that this step's verification URL is not essential for completing the task.

**Comment by han032206** [comment]
Hi there, thanks for the feedback! Really apologies for the late response. We’ve received similar feedback from the community, and have made some modification of our mind2web-live dataset. Currently, we updated the task instructions to be more specific, such as explicitly instruct the agent to perform searches only on certain websites. However, we still encounter cases where multiple paths can lead to the completion of the task, which, as you pointed out, is not fully captured by the current evaluation method.

Ideally, such cases would require multiple key node sequences for evaluation to enable more accurate in progress evaluation, but we haven’t yet implemented this functionality. We're still refining this work, and we’re open to discussing more robust and accurate approaches for web agent online evaluation. Looking forward to collaborating on finding better solutions together.

**Comment by minghchen** [comment]
Thank you for your reply. I'd like to know if your changes, "We updated the task instructions to be more specific," have been put on [the Challenges of your official website](https://www.imean.ai/web-canvas/challenges). Because after using the dataset_io.py download today, I found that the file is the same as before.

**Comment by han032206** [comment]
> Thank you for your reply. I'd like to know if your changes, "We updated the task instructions to be more specific," have been put on [the Challenges of your official website](https://www.imean.ai/web-canvas/challenges). Because after using the dataset_io.py download today, I found that the file is the same as before.

I was saying that we modified other community reported tasks in this manner, but this one seems a little bit wierd to do so. We have updated the task title to be "Go to knols and search for best selling black hoodies in mens size Big and Tall that is between $25 and $50" for now. Inspired by your feedback, maybe a better way is to set a starting page rather than leave it blank page and remove all key nodes related to stepping on the target website. We will consider implement it in the future.
